app/crud/geostore.py

```diff
@@ -115,3 +132,253 @@ async def create_user_area(geometry: Geometry) -> Geostore:
         geostore = await get_gfw_geostore_from_any_dataset(geo_id)
 
     return geostore
+
+
+async def get_admin_boundary_list(
+    admin_provider: str, admin_version: str
+) -> AdminListResponse:
+    dv: Tuple[str, str] = await admin_params_to_dataset_version(
+        admin_provider, admin_version
+    )
+    dataset, version = dv
+
+    src_table: Table = db.table(version)
+    src_table.schema = dataset
+
+    # What exactly is served-up by RW? It looks like it INTENDS to just
+    # serve admin 0s, but the response contains much more
+    where_clause: TextClause = db.text("adm_level=:adm_level").bindparams(adm_level="0")
+
+    gadm_admin_list_columns: List[Column] = [
+        db.column("adm_level"),
+        db.column("gfw_geostore_id"),
+        db.column("gid_0"),
+        db.column("country"),
+    ]
+    sql: Select = (
+        db.select(gadm_admin_list_columns)
+        .select_from(src_table)
+        .where(where_clause)
+        .order_by("gid_0")
+    )
+
+    rows = await get_all_rows(sql)
+
+    return AdminListResponse.parse_obj(
+        {
+            "data": [
+                {
+                    "geostoreId": str(row.gfw_geostore_id),
+                    "iso": str(row.gid_0),
+                    "name": str(row.country),
+                }
+                for row in rows
+            ],
+        }
+    )
+
+
+async def get_all_rows(sql: Select):
+    rows = await db.all(sql)
+
+    return rows
+
+
+async def get_first_row(sql: Select):
+    row = await db.first(sql)
+
+    return row
+
+
+async def get_gadm_geostore(
```

Comment:
You mentioned you might be working on this, but either now or future PR, might be good to break this up into a few subfunctions to make it more readable (e.g. `add_simplify`, `add_admin_id_filter`, etc.

Reply:
Yeah, it's really big. Will factor into some smaller functions in an isolated commit.